modules/ApiManagement/service/main.bicep

```diff
@@ -111,16 +111,18 @@ param zones array = []
 
 @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource.')
 @allowed([
-  'allLogs'
-  'GatewayLogs'
+  []
```

Comment:
Diagnostic settings are implemented consistently across all modules supporting them.

@clarked-msft could you please open an issue in the board to propose the reasons behind the change for the team to discuss and if accepted to be implemented across the library?
I'd suggest to remove the diagnostic settings change from the scope of this PR and keep the `apiType` allowed values update.

Reply:
I've removed the diagnostic settings change and will open an issue.

Reply:
Thanks @clarked-msft. Looks good. 
Please solve merge conflicts and I'll go ahead approving the PR.

Reply:
@clarked-msft description should be updated too as per suggestion and readme regenerated after that. Sorry if I missed it in the first review.